keps/sig-apps/3140-TimeZone-support-in-CronJob/README.md

```diff
@@ -0,0 +1,509 @@
+
+**Note:** Generally we also wait at least two releases between beta and
+GA/stable, because there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+### Upgrade / Downgrade Strategy
+
+- Upgrades
+
+When upgrading from a release without this feature to a release with `TimeZone`
+users should not notice any change in behavior. The default value for `TimeZone`
+is nil, which is equal to current behavior for backwards compatibility.
+
+- Downgrades
+
+When downgrading from a release with this feature, to a release without `TimeZone`
+there are a few cases:
+  1. If TimeZone feature gate was enabled and user specified a TimeZone, the newly
+  created Jobs after downgrade will return to previous behavior, as if no TimeZone
+  was ever specified.
+  2. If TimeZone feature gate was enabled and user did not specify TimeZone, there
+  should be no change in behavior.
+  3. If TimeZone feature gate was not enabled there should be no change in behavior.
+
+### Version Skew Strategy
+
+This feature has no node runtime implications.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: CronJobTimeZone
+  - Components depending on the feature gate: kube-apiserver, kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+No, the default behavior is maintained independently of the feature gate.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled. The outcome of disabling will depend if the
+new field was set by the user:
+
+1. If there was no value set in `TimeZone` field there will be no change in behavior.
+2. If the user set a valid `TimeZone`, the newly created Jobs will be triggered
+as if the field was never set.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The controller will start reading the new field.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, both units and integration tests for enablement, disablement and transitions.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
```

Comment:
An upgrade flow can be vulnerable to the enable, disable, enable if you have a lease that is acquired by a new KCM, then an old KCM, then a new KCM.  The unit and integration tests need to be robust.

Reply:
ack, added that here